pkg/kubelet/deviceplugin/manager.go

```diff
@@ -74,15 +74,25 @@ func removeContents(dir string) error {
 		return err
 	}
 	for _, name := range names {
-		// TODO: skip checkpoint file and check for file type.
-		err = os.RemoveAll(filepath.Join(dir, name))
+		filePath := filepath.Join(dir, name)
+		if filePath == m.CheckpointFile() {
+			continue
+		}
+		if stat, err := os.Stat(filePath); err == nil && stat.IsDir() {
+			continue
+		}
+		err = os.RemoveAll(filePath)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func (m *ManagerImpl) CheckpointFile() string {
+	return filepath.Join(m.socketdir, "checkpoint")
```

Comment:
Checkpoint is too generic. Can we name it `kubelet_internal_checkpoint`?

Reply:
done.